server/src/test/java/keywhiz/service/resources/automation/v2/SecretResourceTest.java

```diff
@@ -275,6 +277,87 @@
     assertThat(s3).doesNotContain("secret17");
   }
 
+  @Test public void secretVersionListing_notFound() throws Exception {
+    Request put = clientRequest("/automation/v2/secrets/non-existent/versions/0-0").build();
+    Response httpResponse = mutualSslClient.newCall(put).execute();
+    assertThat(httpResponse.code()).isEqualTo(404);
+  }
+
+  @Test public void secretVersionListing_success() throws Exception {
+    int totalVersions = 6;
+    int sleepInterval = 1000; // Delay so secrets have different creation timestamps
```

Comment:
to me it would make more sense to set the timestamps explicitly rather than waiting like this.